python/ray/_private/gc_collect_manager.py

```diff
@@ -0,0 +1,64 @@
+import gc
+import logging
+import threading
+import time
+from typing import Callable, Optional
+
+logger = logging.getLogger(__name__)
+
+
+class PythonGCThread(threading.Thread):
+    """A background thread that triggers Python garbage collection.
+    This thread waits for GC events from CoreWorker and triggers `gc.collect()` when
+    requested, ensuring that collections are spaced out by at least
+    `min_interval` seconds."""
+
+    def __init__(
+        self, min_interval: int = 5, gc_collect_func: Optional[Callable] = None
+    ):
+        logger.debug("Starting Python GC thread")
+        super().__init__(name="PythonGCThread", daemon=True)
+        self._running = True
```

Comment:
```suggestion
        self._should_exit = False
```
nit: a little clearer IMO